src/schema/schema.py

```diff
@@ -98,6 +104,25 @@ class ChatMessage(BaseModel):
         description="Original LangChain message in serialized form.",
         default={},
     )
+    task_name: str | None = Field(
```

Comment:
to clean up ChatMessage, we could add a Task class with these fields and ChatMessage could have a task optional field.

Reply:
This looks appealing, but I think it breaks the proposed abstraction.

`Task` class models a background task. As the task progresses (when it starts, finishes, or when it produces data during its execution), it emits TaskMessages, which (similar to how HumanMessages or AIMessages are handled) are then converted to ChatMessages, which serve as universal communication tool between server and client.

That said, I believe placing `Task` itself into `ChatMessage` is wrong, as this way `ChatMessage` would contain a reference to background Task, which can progress during ChatMessage's lifetime. This contradicts ChatMessage's purpose to communicate a single notification produced by Task execution at given point in time.